assets/src/editor/events/initialization/use-editor-persistence.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External imports
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import jquery from 'jquery';
+
+/**
+ * A custom hook connecting with the update/publish submit action in a WordPress
+ * post editor that will persist any dirty store state for the `eventespresso/core`
+ * store when submit is triggered by  user.
+ *
+ * @param {number} eventId  The eventId (post id) for the current editor
+ *                          instance.  Currently used for breaking cache on
+ *                          callbacks.
+ */
+const useEditorPersistence = ( eventId ) => {
+	const eventProcessed = useRef( false );
+	const {
+		persistAllDeletes,
+		persistAllAddRelations,
+		persistAllDeleteRelations,
+		persistAllEntities,
+	} = useDispatch( 'eventespresso/core' );
+	const persistAll = useCallback( async ( event ) => {
+		if ( eventProcessed.current === false ) {
+			event.preventDefault();
+			await persistAllDeletes();
+			await persistAllAddRelations();
+			await persistAllDeleteRelations();
+			await persistAllEntities();
```

Comment:
wouldn't you need to call `persistAllEntities()` before `persistAllAddRelations()` and `persistAllDeleteRelations()` so that any newly created entities have the correct ID?

Reply:
The relation persisting automatically takes care of persisting any dirty entities first before persisting the relations, so the `persistAllEntities` here is just to catch any entities that have no relations.  However, reflecting on your comment, in the context of the current state of the editor, it will likely be more performant if we persist all entities before doing the relation persisting so I'll do that (I do think I need to still fix the bug reported in "known issues" around relation connection before doing this though so it doesn't surface in other contexts).

Should note also, if you find out this logic works okay, you can probably create an aggregate action generator in the store for handling this kind of `persistAllState` behaviour.